hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleEntityId.java

```diff
@@ -106,20 +104,14 @@ public boolean equals(Object o) {
 		}
 
 		var that = (MerkleEntityId)o;
-		return new EqualsBuilder()
-				.append(shard, that.shard)
-				.append(realm, that.realm)
-				.append(num, that.num)
-				.isEquals();
+		return this.shard == that.shard && this.realm == that.realm && this.num == that.num;
 	}
 
 	@Override
 	public int hashCode() {
-		return new HashCodeBuilder(17, 37)
-				.append(shard)
-				.append(realm)
-				.append(num)
-				.toHashCode();
+		int result = Long.hashCode(shard);
+		result = 31 * result + Long.hashCode(realm);
+		return 31 * result + Long.hashCode(num);
```

Comment:
Adding a modulo of Integer.MAX_VALUE would avoid the case of integer overflow, for eg
`result = ((31 * result) % Integer.MAX_VALUE + Long.hashCode(realm)) % Integer.MAX_VALUE;`

Reply:
The `hashCode()` method should return values randomly sampled across the _entire range_ of `int` values. So overflow is not only ok, it's desirable. 😁

Reply:
I see, thanks for clarifying.